packages/engine/Source/Core/FpsTracker.js

```diff
@@ -0,0 +1,40 @@
+import destroyObject from "../Core/destroyObject.js";
+import getTimestamp from "../Core/getTimestamp.js";
+
+/**
+ * @private
+ */
+function FpsTracker(options) {
+  this._timeStamps = [];
+}
+
+/**
+ * Calculates average fps based on timeStamps of recently rendered frames
+ */
+FpsTracker.prototype.getAverageFps = function () {
```

Comment:
The hard-coded `10` (and the corresponding `9`s below) make my eye twitch 🙁 

But beyond that: This does not really compute the "average FPS". 

Consider time stamps like
`1 2 3 4 5 6 7 8 9 10`
It will compute `(10 - 1) / 9 = 1.0`. 
This means that the ~"time per frame" is `1.0`. (This isn't really the "elapsed time", but ... well...)

For
`1 2 3 4 5 6 7 8 9 19`
it will compute `(19 - 1) / 9 = 2.0`.
So even though the "time per frame" will constantly be `1.0` (except for the outlier of the last frame), the "time per frame" will be computed as `2.0`. The whole idea behind storing multiple time stamps is _exactly_ to "smooth (average) out" single outliers like that last `19`. 

One could mention that, in the current approach, it wouldn't even be necessary to store the 10 time stamps. Storing **2** (and a 'counter') would be enough. For a proper "sliding window", one could examine the performance of `unsihift`/`pop` vs. `push`/`shift`. Or one avoid these considerations altogether and use a fixed-sized array and a 'cursor' (aka "circular buffer").

Reply:
This should be obsolete with the `FrameRateMonitor`. 

(While some aspects of this class are hard to understand at a glimpse (like that "visibility" thing....), it seems to offer the required functionality, so _not_ duplicating that (even in a simpler, more self-contained form) seems reasonable)